hedera-node/hedera-app/src/main/java/com/hedera/node/app/state/merkle/disk/OnDiskMerkleDbValueSerializer.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.node.app.state.merkle.disk;
+
+import com.hedera.node.app.state.merkle.StateMetadata;
+import com.hedera.pbj.runtime.Codec;
+import com.hedera.pbj.runtime.io.buffer.BufferedData;
+import com.swirlds.merkledb.serialize.ValueSerializer;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.util.Objects;
+
+public class OnDiskMerkleDbValueSerializer<V> implements ValueSerializer<OnDiskValue<V>> {
+
+    @Deprecated(forRemoval = true)
```

Comment:
This isn't deprecated, as there is no replacement.  Also, there is no reason ever to deprecate a private variable (there is nothing outside the class using it).  Note that it needs to be replace in a comment, perhaps, but annotating it deprecated just creates compiler noise.

Reply:
I didn't introduce it, it was copied from the old JasperDB serializer class, where it was introduced by @rbair23 I assume the annotation was just to indicate that it eventually needs to be cleaned up.

Reply:
Not strictly required, of course, but we should be cleaning these up as we see them.
There are many hundreds of deprecation warnings in compiler output for things that have no alternative or are private, so cleaning those up helps with finding real issues by reducing useless warnings.

Reply:
All I can do for now is to remove the annotation, still leaving comments that eventually it needs to be refactored. There is no real alternative to ConstructableRegistry and to the way how it creates objects.

Reply:
That is precisely the suggestion, remove the annotation for private values or anything that has no alternative; since it creates warnings we cannot resolve.